client/src/pages/Review/Review.tsx

```diff
@@ -58,86 +68,35 @@ const Review = () => {
     return () => window.removeEventListener("beforeunload", beforeUnload);
   }, []);
 
-  useEffect(() => {
-    if (currentCourseKey === null) {
-      setTimeout(() => history.pushState(null, "", "/app"), 10);
-    } else {
-      setTimeout(() => history.pushState(null, "", `/app/${currentCourseKey}`), 10);
-    }
-  }, [currentCourseKey]);
```

Comment:
This `useEffect` monitoring changes on `currentCourseKey` so the url is updated as soon as the current course key changes.
 
This is here so that developers don't have to always put `window.history.pushState` under every single `setCurrentCourseKey` statement, because it's repetitive and forgettable.

Reply:
I see. What's the timeout for?

Reply:
You can remove `timeout` now. It was used to avoid race condition but it doesn't seem to have it anyways.